HW_9/task1.py

```diff
@@ -0,0 +1,68 @@
+# 1. Напишите следующие функции:
+#    * Нахождение корней квадратного уравнения
+#    * Генерация csv файла с тремя случайными числами в каждой строке. 100-1000 строк.
+#    * Декоратор, запускающий функцию нахождения корней квадратного уравнения с каждой тройкой чисел из csv файла.
+#    * Декоратор, сохраняющий переданные параметры и результаты работы функции в json файл.
+
+import json
+import csv
+import math
+import os
+import random
+
+
+def gen_csv(count=3, start=1, stop=100, rows_min=100, rows_max=1000):
+    with open("random_num.csv", "w", encoding="utf-8", newline="") as file_csv:
+        writer = csv.writer(file_csv, delimiter=",", quoting=csv.QUOTE_NONNUMERIC)
+        writer.writerows(
+            [[random.randint(start, stop) for _ in range(count)] for _ in range(random.randint(rows_min, rows_max))]
+        )
+
+
+def read_csv(func):
+    gen_csv()
+
+    def wrapper():
+        if os.path.exists("random_num.csv"):
+            with open("random_num.csv", "r", encoding='utf-8', newline="") as file_csv:
+                for item in csv.reader(file_csv, quoting=csv.QUOTE_NONNUMERIC):
+                    func(*item)
+
+                return func
+        else:
+            return func
```

Comment:
Скорее всего, если файл не существует, а такое вполне может быть, если после декорирования вручную удалить, файл нужно снова его сгенерировать.
```suggestion
        else:
            gen_csv()
```